rotkehlchen/chain/evm/decoding/morpho/utils.py

```diff
@@ -0,0 +1,168 @@
+
+        except (requests.RequestException, JSONDecodeError, TypeError, KeyError) as e:
+            error = f'missing key {e!s}' if isinstance(e, KeyError) else f'{e!s}'
+            log.error(f'Failed to retrieve morpho vaults from api due to {error}')
+            return None
+
+    return all_vaults
+
+
+def query_morpho_vaults(database: 'DBHandler') -> None:
+    """Query list of Morpho vaults and add the vault tokens to the global database."""
+    with GlobalDBHandler().conn.read_ctx() as cursor:
+        last_vault_count = globaldb_get_unique_cache_value(
+            cursor=cursor,
+            key_parts=(CacheType.MORPHO_VAULTS,),
+        )
+
+    if (vault_list := _query_morpho_vaults_api()) is None:
+        _update_cache_timestamp()  # Update timestamp to prevent repeated errors.
+        return
+
+    _update_cache_timestamp(count=(vault_count := len(vault_list)))
+
+    try:
+        if last_vault_count is not None and vault_count == int(last_vault_count):
+            log.debug(f'Same number ({vault_count}) of Morpho vaults returned from API as previous query. Skipping vault processing.')  # noqa: E501
+            return
+    except ValueError:
+        log.error(f'Failed to check last Morpho vault count due to {last_vault_count} not being an int')  # noqa: E501
+        return
+
+    for vault in vault_list:
+        try:
+            vault_chain_id = ChainID.deserialize_from_db(vault['chain']['id'])
+            underlying_token = get_or_create_evm_token(
+                userdb=database,
+                evm_address=string_to_evm_address(vault['asset']['address']),
+                chain_id=vault_chain_id,
+                decimals=deserialize_int(vault['asset']['decimals']),
+                name=vault['asset']['name'],
+                symbol=vault['asset']['symbol'],
+                encounter=TokenEncounterInfo(description='Querying Morpho vaults', should_notify=False),  # noqa: E501
+            )
+            get_or_create_evm_token(
+                userdb=database,
+                evm_address=string_to_evm_address(vault['address']),
+                chain_id=vault_chain_id,
+                protocol=MORPHO_VAULT_PROTOCOL,
+                decimals=18,  # all morpho vaults have 18 decimals
+                name=vault['name'],
+                symbol=vault['symbol'],
+                underlying_tokens=[UnderlyingToken(
+                    address=underlying_token.evm_address,
+                    token_kind=EvmTokenKind.ERC20,
+                    weight=ONE,
+                )],
+                encounter=TokenEncounterInfo(description='Querying Morpho vaults', should_notify=False),  # noqa: E501
+            )
+        except (DeserializationError, KeyError) as e:
+            error = f'missing key {e!s}' if isinstance(e, KeyError) else f'{e!s}'
+            log.error(f'Failed to store token information for Morpho vault due to {error}. Vault: {vault}. Skipping...')  # noqa: E501
```

Comment:
here you forgot to log which chain it is in. Add chain name

```suggestion
            log.error(f'Failed to store token information for Morpho vault due to {error}. Vault: {vault}. Skipping...')  # noqa: E501
```

Reply:
This is processing morpho vaults from all chains. While we could use `vault['chain']['id']` to get the chain for the current vault here, deserializing that value is one of the things that could raise this error in the first place. 

Also the vault data is already being logged with `Vault: {vault}`, which will include the vault's chain id and name.